lib/internal/modules/esm/fetch_module.js

```diff
@@ -44,44 +44,67 @@ const cacheForGET = new SafeMap();
 // [2] Creating a new agent instead of using the gloabl agent improves
 // performance and precludes the agent becoming tainted.
 
+/** @type {import('https').Agent} The Cached HTTP Agent for **secure** HTTP requests. */
 let HTTPSAgent;
-function HTTPSGet(url, opts) {
+/**
+ * Make a **secure** HTTP GET request (handling agent setup if needed, caching the agent to avoid
+ * redudant instantiations).
+ * @param {Parameters<import('https').get>[0]} input - The URI to fetch.
+ * @param {Parameters<import('https').get>[1]} options - See https.get() options.
+ */
+function HTTPSGet(input, options) {
   const https = require('https'); // [1]
   HTTPSAgent ??= new https.Agent({ // [2]
     keepAlive: true,
   });
-  return https.get(url, {
+  return https.get(input, {
     agent: HTTPSAgent,
-    ...opts,
+    ...options,
   });
 }
 
+/** @type {import('https').Agent} The Cached HTTP Agent for **insecure** HTTP requests. */
 let HTTPAgent;
-function HTTPGet(url, opts) {
+/**
+ * Make a **insecure** HTTP GET request (handling agent setup if needed, caching the agent to avoid
```

Comment:
What's insecure about it?
```suggestion
 * Make a HTTP GET request (handling agent setup if needed, caching the agent to avoid
```